pyo3-macros-backend/src/pymethod.rs

```diff
@@ -634,31 +634,28 @@ pub fn impl_py_getter_def(
                 span: Span::call_site(),
             }
             .receiver(cls, ExtractErrorMode::Raise);
-            if let Some(ident) = &field.ident {
+            let field_token = if let Some(ident) = &field.ident {
                 // named struct field
-                quote!(::std::clone::Clone::clone(&(#slf.#ident)))
+                ident.to_token_stream()
             } else {
                 // tuple struct field
-                let index = syn::Index::from(field_index);
-                quote!(::std::clone::Clone::clone(&(#slf.#index)))
-            }
-        }
-        PropertyType::Function {
-            spec, self_type, ..
-        } => impl_call_getter(cls, spec, self_type)?,
-    };
-
-    let conversion = match property_type {
-        PropertyType::Descriptor { .. } => {
+                syn::Index::from(field_index).to_token_stream()
+            };
             quote! {
-                let item: _pyo3::Py<_pyo3::PyAny> = _pyo3::IntoPy::into_py(item, _py);
-                ::std::result::Result::Ok(_pyo3::conversion::IntoPyPointer::into_ptr(item))
+                ::std::result::Result::Ok(
+                    _pyo3::conversion::IntoPyPointer::into_ptr(
+                        _pyo3::IntoPy::<_pyo3::Py<_pyo3::PyAny>>::into_py(::std::clone::Clone::clone(&(#slf.#field_token)), _py)
```

Comment:
To be honest, I think this somewhat showcases the downside of the expression style insofar this expression is becoming an unreadable wall of text. The previous version with the extra binding was not great, but still more readable IMHO.

Could we at least try to use interpolation and multiple `quote!` invocation to try to pull this apart in our macro implementation while still emitting a single expression in the end?

Reply:
Sure thing, I'll push a refactoring later (might create a bit of a bigger diff).